framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceScenes.java

```diff
@@ -477,6 +485,38 @@ public void updateInferredMethodReturnType(
         AnnotatedTypeMirror rhsATM = atf.getAnnotatedType(retNode.getTree().getExpression());
         helper.updateAnnotationSetInScene(
                 method.returnType, atf, jaifPath, rhsATM, lhsATM, TypeUseLocation.RETURN);
+
+        // Now, update overridden methods based on the implementation we just saw.
+        // If the overridden method is abstract, then WPI cannot update it directly,
+        // so updating based on implementations makes sense (cf. how method parameters
+        // are updated).
+        for (Map.Entry<AnnotatedDeclaredType, ExecutableElement> pair :
+                overriddenMethods.entrySet()) {
+
+            AnnotatedExecutableType overriddenMethod =
+                    AnnotatedTypes.asMemberOf(
+                            atf.getProcessingEnv().getTypeUtils(),
+                            atf,
+                            pair.getKey(),
```

Comment:
Rather than making multiple calls to `getKey` and `getValue`, make one call and assign to a local variable with a mnemonic name.  That will be easier to understand than making the reader remember what `pair` means.